barretenberg/cpp/src/barretenberg/vm2/simulation/sha256_compression.cpp

```diff
@@ -0,0 +1,51 @@
+#include "barretenberg/vm2/simulation/sha256_compression.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <sys/types.h>
+
+#include "barretenberg/vm2/simulation/lib/sha256_compression.hpp"
+
+namespace bb::avm2::simulation {
+
+void Sha256::compression(ContextInterface& context,
+                         MemoryAddress state_addr,
+                         MemoryAddress input_addr,
+                         MemoryAddress output_addr)
+{
+    std::array<uint32_t, 8> state;
+    std::array<uint32_t, 16> input;
+
+    auto& memory = context.get_memory();
+
+    // Load 8 elements representing the state from memory.
+    for (uint32_t i = 0; i < 8; ++i) {
+        auto memory_value = memory.get(state_addr + i).value;
+        /*Check Tag is U32 */
+        state[i] = static_cast<uint32_t>(memory_value);
+    }
+
+    // Load 16 elements representing the input from memory.
+    for (uint32_t i = 0; i < 16; ++i) {
+        auto memory_value = memory.get(input_addr + i).value;
+        /*Check Tag is U32 */
```

Comment:
Is this an error handling TODO? Do you need to check the tags here? if yes please make explicit